src/RequestCompressionMiddleware.php

```diff
@@ -87,8 +85,12 @@ private function compressRequestBody(
         $body = $request->getBody()->getContents();
         $compressedBody = $fn($body);
 
-        return $request->withBody(Psr7\Utils::streamFor($compressedBody))
-            ->withHeader('content-encoding', $this->encoding);
+        if (!$request->hasHeader('Content-Encoding')) {
+            return $request->withBody(Psr7\Utils::streamFor($compressedBody))
+                ->withHeader('Content-Encoding', $this->encoding);
+        }
+
+        return $request->withAddedHeader('Content-Encoding', $this->encoding);
```

Comment:
this appends the compression algorithm to an existing `Content-Encoding` header, if present.  See `SDKAppendedGzipAfterProvidedEncoding_restJson1` in `input/rest-json.json`